src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,665 @@
+        {
+            return new VersionInfo(
+                modelSignature: "TENSFLOW",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        private readonly Bindings _bindings;
+
+        /// <summary>
+        /// Tensorflow session object
+        /// </summary>
+        private TFSession _session;
+
+        public override ISchema Schema => _bindings;
+
+        /// <summary>
+        ///  First dimension in each tensor is a batch dimension.
+        ///  Currently setting it to 1.
+        /// </summary>
+        private readonly int _batchSize;
+
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="modelFile">This is the frozen model file. https://www.tensorflow.org/mobile/prepare_models </param>
+        /// <param name="name">Name of the output column. Keep it same as in the Tensorflow model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
+        public TensorflowTransform(IHostEnvironment env, IDataView input, string modelFile, string name, params string[] source)
+            : this(env, new Arguments() { Column = new[] { new Column() { Source = source, Name = name } }, ModelFile = modelFile }, input)
+        {
+        }
+
+        public TensorflowTransform(IHostEnvironment env, Arguments args, IDataView input)
+            : base(env, RegistrationName, input)
+        {
+            Host.CheckValue(args, nameof(args));
+            Host.CheckUserArg(Utils.Size(args.Column) > 0, nameof(args.Column));
+            for (int i = 0; i < args.Column.Length; i++)
+                Host.CheckUserArg(Utils.Size(args.Column[i].Source) > 0, nameof(args.Column));
+
+            _batchSize = args.BatchSize;
+            _session = LoadTFSession(args.ModelFile);
+            _bindings = new Bindings(args.Column, Source.Schema, this);
+        }
+
+        private TensorflowTransform(IHost host, ModelLoadContext ctx, IDataView input)
+           : base(host, input)
+        {
+            Host.AssertValue(ctx);
+
+            _batchSize = ctx.Reader.ReadInt32();
+
+            byte[] data = null;
+            ctx.TryLoadBinaryStream("TFModel", r =>
+            {
+                data = r.ReadByteArray();
+            });
+
+            var graph = new TFGraph();
+            graph.Import(data);
+            _session = new TFSession(graph);
+            _bindings = new Bindings(ctx, Source.Schema, this);
+        }
+
+        ~TensorflowTransform()
+        {
+            _session.CloseSession();
+            _session.DeleteSession();
```

Comment:
>_session [](start = 12, length = 8)

Should we also call _session.Dispose()?

Reply:
I see dispose already calls _session.DeleteSession();

---
In reply to: [210647126](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210647126) [](ancestors = 210647126)